### PR TITLE
Mailbox 284: Error handling for Attachment parser

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

**Comment by chibenwa** [comment]
First commit do not seem linked with this issue. Please remove.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java [hunk 1]
```diff
@@ -3126,4 +3126,66 @@ public void messageWithNonIndexableAttachmentShouldHaveItsEmailBodyIndexed() thr
...
+            "      }}" +
+            "    }," +
+            "    \"#0\"" +
+            "  ]" +
+            "]";
+
+        String createdPath = ARGUMENTS + ".created[\""+messageCreationId+"\"]";
+        String firstAttachment = createdPath + ".attachments[0]";
+
+        given()
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: Indent here ^^

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java [hunk 2]
```diff
@@ -304,24 +304,29 @@ private MetaDataWithContent createMessageInOutbox(MessageWithId.CreationMessageE
     }
 
     private ImmutableList<MessageAttachment> getMessageAttachments(MailboxSession session, ImmutableList<Attachment> attachments) throws MailboxException {
-        ThrowingFunction<Attachment, MessageAttachment> toMessageAttachment = att -> messageAttachment(session, att);
+        ThrowingFunction<Attachment, Optional<MessageAttachment>> toMessageAttachment = att -> messageAttachment(session, att);
         return attachments.stream()
             .map(Throwing.function(toMessageAttachment).sneakyThrow())
+            .filter(Optional::isPresent)
```
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: .flatMap(OptionalConverter::toStream)

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/impl/MessageParserTest.java [hunk 3]
```diff
@@ -198,4 +198,18 @@ public void getAttachmentsShouldNotConsiderUnknownContentDispositionAsAttachment
 
         assertThat(attachments).hasSize(0);
     }
+
+    @Test
+    public void getAttachmentsShouldRetrieveAttachmentsWhenOneFailOnWrongContentDisposition() throws Exception {
```
Comment (aduprat) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/impl/MessageParserTest.java]: I think those two cases need to be in different commits.

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java [hunk 4]
```diff
@@ -91,7 +94,13 @@
                 attachments.addAll(listAttachments((Multipart) entity.getBody(), Context.fromEntity(entity)));
             } else {
                 if (isAttachment(entity, context)) {
-                    attachments.add(retrieveAttachment(messageWriter, entity));
+                    try {
+                        attachments.add(retrieveAttachment(messageWriter, entity));
```
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java]: I think _retrieveAttachment_ should not throw when:
- inline attachment without cid
- unknown content-disposition

Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java]: Different tickets on the biard

**Comment by chibenwa** [comment]
Have just been merged
